### PR TITLE
Add support for Capacitor 4.0.0

### PR DESCRIPTION
**Pull request by sawaca96**
capacitor 4.0.0 deployed! ([release](https://github.com/ionic-team/capacitor/releases))

so fixes `error: compiling for iOS 12.0, but module 'Capacitor' has a minimum deployment target of iOS 13.0`

### CODE REVIEW

**Comment by lucas-zimerman** [comment]
Hi @sawaca96, Was the bump on the minimum android version also required in order to run with Capacitor 4?

**Comment by sawaca96** [comment]
I think bump up android version is not necessary

File: SentryCapacitor.podspec [hunk 1]
```diff
@@ -11,7 +11,7 @@ Pod::Spec.new do |s|
   s.author = package['author']
   s.source = { :git => package['repository']['url'], :tag => s.version.to_s }
   s.source_files = 'ios/Plugin/**/*.{swift,h,m,c,cc,mm,cpp}'
-  s.ios.deployment_target  = '12.0'
+  s.ios.deployment_target  = '13.0'
```
Comment (lucas-zimerman) [inline SentryCapacitor.podspec:14]: My main concern here is  users using the SDK targeting iOS 12.0, on the best scenario we could target 12.0 for older Ionic versions and 13 for the latest version. What do you think about it?

Comment (sawaca96) [reply SentryCapacitor.podspec:14]: sounds nice 😊 

I wonder how to know the capacitor version

Comment (lucas-zimerman) [inline SentryCapacitor.podspec:14]: You can check using the following code

```ruby
capacitorPackage =  JSON.parse(File.read(File.join(__dir__, '../../@capacitor/core/package.json')))
capacitorVersion = capacitorPackage['version']
if capacitorVersion.start_with?("2.") or capacitorVersion.start_with?("3.")
  miniOSVersion = '12.0'
else 
  miniOSVersion = '13.0' # Required for Capacitor 4 and newer.
end```

```ruby
  s.ios.deployment_target  = miniOSVersion

```

I believe with that we can keep the support for old devices as long as people use the old Capacitor.

**Comment by lucas-zimerman** [comment]
Also, could you update the `changelog.md` with the following message?

```
## Unreleased

### Features

- Add support for Capacitor 4.0.0 ([#201](https://github.com/getsentry/sentry-capacitor/pull/201))

### Fixes

- Bump Sentry Javascript to fix incompatibility with Sentry Tracing ([#202](https://github.com/getsentry/sentry-capacitor/pull/202))
```

File: SentryCapacitor.podspec (hunk 1 above)
Comment (piotr-cz) [reply SentryCapacitor.podspec:14]: How about leaving the `deployment_target` on 12.0 and relying on `@capacitor/core` to check if it's requirement is met?
According to [migration docs](https://capacitorjs.com/docs/updating/4-0#raise-ios-deployment-target) it should be built in `ios/App/Podfile`:

```
post_install do |installer|
  assertDeploymentTarget(installer)
end
```

**Comment by piotr-cz** [comment]
This PR doesn't update devDependencies of `@capacitor/*` packages in [package.json](https://github.com/getsentry/sentry-capacitor/blob/main/package.json).

IMHO these should be changed to
```json
{
  "devDependencies": {
    "@capacitor/android": "^3.0.1 || ^4.0.0",
    "@capacitor/core": "^3.0.1 || ^4.0.0",
    "@capacitor/ios": "^3.0.1 || ^4.0.0"
  }
}
```

**Comment by lucas-zimerman** [comment]
Hi @sawaca96, If you don't mind, I'll merge this PR on another branch and add the requested changes on it :D

**Comment by sawaca96** [comment]
> Hi @sawaca96, If you don't mind, I'll merge this PR on another branch and add the requested changes on it :D

Sorry for late...😂

- a182cd4 about comment(is it right?)
- 27e6d7a merge commit

**Comment by sawaca96** [comment]
0c0b2b2

I added the missing setting referring to [guide](https://capacitorjs.com/docs/updating/4-0).

**Review by lucas-zimerman — APPROVED** [review]
Thank you for the Pull Request!

**Pull request merged**